sdk/communication/AzureCommunicationChat/Source/PushNotification/PushNotificationEncryptionDelegate.swift

```diff
@@ -0,0 +1,33 @@
+// --------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation. All rights reserved.
+//
+// The MIT License (MIT)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the ""Software""), to
+// deal in the Software without restriction, including without limitation the
+// rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//
+// --------------------------------------------------------------------------
+
+import Foundation
+
+public protocol PushNotificationEncryptionDelegate {
+    func store(encryptionKey: String, expiryTime: Date)
+
+    func getEncryptionKeys() -> [String]?
```

Comment:
Delegates must follow the guidelines for delegates:
**https://azure.github.io/azure-sdk/ios_design.html#delegates**

Reply:
Thanks for sharing. I'm reading it right now.

Reply:
I just read through the example in this wiki and did some research on "Delegate" in swift. 
For me "PushNotificationAdapter" is just like an interface and don't need to be specifically defined as a "delegate".
Could you please provide a reason on why we should treat it as a "Delegate"? @tjprescott